mkosi/__init__.py

```diff
@@ -823,6 +833,62 @@ def install_systemd_boot(state: MkosiState) -> None:
                      state.workspace / "mkosi.esl"])
 
 
+def install_shim(state: MkosiState) -> None:
+    if not want_efi(state.config):
+        return
+
+    if state.config.shim_bootloader == ShimBootloader.none:
+        return
+
+    if not any(gen_kernel_images(state)) and state.config.bootable == ConfigFeature.auto:
+        return
+
+    signed = [
+        "usr/lib/shim/shim*.efi.signed.latest",
+        "usr/lib/shim/shim*.efi.signed",
+        "boot/efi/EFI/*/shim[a-z][0-9]*.efi",
+        "boot/efi/EFI/*/shim.efi",
+        "usr/lib64/efi/shim-*signed.efi",
+        "usr/share/efi/*/shim-*signed.efi",
+    ]
+
+    unsigned = [
+        "usr/share/shim/shim[a-z]*[0-9]*.efi",
+        "usr/share/shim/*/*/shim[a-z]*[0-9]*.efi",
+        "usr/share/efi/*/shim.efi",
+        "usr/lib64/efi/shim.efi",
+        "usr/lib/shim/shim[a-z]*[0-9]*.efi",
+    ]
+
+    dst = efi_boot_binary(state)
+    with umask(~0o700):
+        (state.root / dst).parent.mkdir(parents=True, exist_ok=True)
+
+    if state.config.shim_bootloader == ShimBootloader.signed:
+        for pattern in signed:
+            for p in state.root.glob(pattern):
+                log_step(f"Installing presigned shim EFI binary from /{p.relative_to(state.root)} to /{dst}")
+                shutil.copy2(p, state.root / dst)
+                return
```

Comment:
Is there any possibility that a single glob could match multiple multiple files? Does it make sense to copy only the first one in that case and if so, should this maybe get some sort of version sort?

Reply:
None of the distribution packages I've looked at ship names with versions in them so I wouldn't bother for now